Presentation: Learning snippets cleanup

grigasp opened the pull request.

roluk reviewed (changes requested):
Please also delete reference to a non-existent `Backend.Initialization.RpcInterface` snippet from `PresentationRpcImpl.ts`.

Also, something needs to be done with snippet located at https://www.itwinjs.org/presentation/advanced/testing/

The code comments there refers to a VSTS item, has an ESLint disable comment, and a test that is skipped.

grigasp commented:
> Please also delete reference to a non-existent `Backend.Initialization.RpcInterface` snippet from `PresentationRpcImpl.ts`.
> 
> Also, something needs to be done with snippet located at https://www.itwinjs.org/presentation/advanced/testing/
> 
> The code there refers to a VSTS item, has an ESLint disable comment, and a test that is skipped.

Cleaned up `PresentationRpcImpl.ts` docs, reworked `RulesetTesting` tests and docs. Please have a look.

roluk commented on docs/presentation/Advanced/Testing.md:
```diff
@@ -1,33 +1,51 @@
-1. resulting hierarchies get rather large - testing the them in
-code might be difficult
-2. snapshots protect against regressions
+1. Resulting hierarchies get rather large - testing the them in code might be difficult.
```
```suggestion
1. Resulting hierarchies get rather large — testing them in code might be difficult.
```

roluk commented on docs/presentation/Advanced/Testing.md:
````diff
@@ -1,33 +1,51 @@
+    [[include:Presentation.Testing.Rulesets.Terminate]]
+    ```
+
+**Note:** The above example uses [mocha](https://www.npmjs.com/package/mocha) and [chai-jest-snapshot](https://www.npmjs.com/package/chai-jest-snapshot) packages to perform snapshot testing. The path returned by `createSnapshotPath` function used in the tests should match the project structure and is outside the scope of this example.
````
> The path returned by `createSnapshotPath` function used in the tests should match the project structure [..]

Does chai-jest-snapshot really care where you place snapshot files? I think you should omit this sentence.

grigasp replied:
I'm using this function in the example code and now including its content. I think it makes sense to tell what it's for.

roluk replied:
Does chai-jest-snapshot really care where you place snapshot files though? I find this sentence potentially misleading.

roluk replied:
Then let's not mislead people that a particular file structure is required for snapshots to work... What `createSnapshotPath` does is obvious from the name, but if you insist on over explaining it, you can do it without imposing our internal requirements to the end users:

```suggestion
**Note:** The above example uses [mocha](https://www.npmjs.com/package/mocha) and [chai-jest-snapshot](https://www.npmjs.com/package/chai-jest-snapshot) packages to perform snapshot testing. `createSnapshotPath` function retrieves a path to the snapshot for the current test.
```
